src/main/webapp/script.js

```diff
@@ -12,7 +12,118 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+/**
+ * Submits a fetch request to the /data url. Upon receiving a JSON encoding of the
+ * nodes and edges of the graph, renders the graph in a container on the page using
+ * the cytoscape.js library
+ */
+async function generateGraph() {
+  // Arrays to store the cytoscape graph node and edge objects
+  let graphNodes = [];
+  let graphEdges = [];
+  const response = await fetch("/data");
 
-function testFunc() {
-  fetch("/data").then(response => true);
-}
\ No newline at end of file
+  const serverErrorStatus = response.headers.get("serverError");
+
+  // Error on server side
+  if (serverErrorStatus !== null) {
+    displayError(serverErrorStatus);
+  }
+
+  const jsonResponse = await response.json();
+  // Graph nodes and edges received from server
+  let nodes = jsonResponse[0];
```

Comment:
Instead of hardcoding index [0] and [1] here, it seems less error prone to specify the Json data as {"nodes": nodeJson, "edges": edgeJson}

Reply:
So I'm having a bit of trouble figuring out the Json string. 

The first time I forgot the \" around nodes and the second time I forgot the \" around the nodeJson/edgeJson string. Currently I have: `String bothJson = "{\"nodes\": " + "\"" + nodeJson + "\""  + "," + "\"edges\": " + "\"" + edgeJson + "\"" + "}";` (which I can clean up a bit)

The toJson on the client works if I do something like
` String nodeJson = "hi";
 String edgeJson = "bye";`
but it doesn't with the actual nodeJson and edgeJson strings. Do you know why this could be the case?

Reply:
Just kidding, I used the JSONObject library to construct the object instead of doing it manually!

Reply:
do you think we still need to do tests for JSON if we use a library?

Reply:
Having a small test doesn't hurt :) It doesn't have to be in this PR though.